backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -831,4 +837,22 @@ public FutureExperimentRunDAO(
             ModelDBActionEnum.ModelDBServiceActions.UPDATE)
         .thenCompose(unused -> createExperimentRunHandler.createExperimentRun(request), executor);
   }
+
+  public InternalFuture<Void> logVersionedInputs(LogVersionedInput request) {
+    final var runId = request.getId();
+    final var now = Calendar.getInstance().getTimeInMillis();
+    if (!request.hasVersionedInputs()) {
+      throw new InvalidArgumentException("VersionedInput not found in request");
```

Comment:
This check should be inside `validateAndInsertVersionedInputs`

Reply:
Moved it as per suggestion